plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -214,13 +217,27 @@ public static Schema constructSchema(Schema dataSchema, List<String> columnNames
             }
 
             fieldBuilder = fieldBuilder
-                    .name(columnName)
+                    .name(field.name())
                     .type(typeForNewField)
                     .withDefault(null);
         }
         return fieldBuilder.endRecord();
     }
 
+    public static Schema.Field getFieldFromSchema(String columnName, Schema schema)
+    {
+        Schema.Field field = schema.getField(columnName);
+        if (field != null) {
+            return field;
+        }
+
+        return schema.getFields().stream()
+                .filter(f -> f.name().equalsIgnoreCase(columnName))
+                .findFirst()
+                .orElseThrow(() -> new TrinoException(HUDI_INVALID_SCHEMA_ERROR,
+                        "Failed to get column " + columnName + " from table schema"));
+    }
```

Comment:
columnName is case insensitive, while field.Name is case sensitive right? 

Let's include a javadoc here to explain why this is here, explaining that `columnName` is resolved form HMS or a catalog, where column names are synced in a case insensitive way.